quixstreams/sources/base/manager.py

```diff
@@ -79,6 +111,46 @@ def run(self) -> None:
             "s" if threadcount > 1 else "",
         )
 
+    def _recover_state(self, source: StatefulSource) -> StorePartition:
+        """
+        Recover the state from the changelog topic and return the assigned partition
+
+        For stateful sources only.
+        """
+        recovery_manager = RecoveryManager(
+            consumer=self._consumer,
+            topic_manager=self._topic_manager,
+        )
+
+        state_manager = StateStoreManager(
+            producer=self._producer, recovery_manager=recovery_manager
+        )
+
+        store_name = f"source-{source.store_name}"
+        state_manager.register_store(
+            topic_name=None,
+            store_name=store_name,
+            store_type=MemoryStore,
+            topic_config=TopicConfig(
+                num_partitions=source.store_partitions_count,
+                replication_factor=self._topic_manager.default_replication_factor,
+            ),
+        )
+
+        self._topic_manager.create_all_topics()
+        self._topic_manager.validate_all_topics()
+
+        store_partitions = state_manager.on_partition_assign(
+            topic=None,
+            partition=source.assigned_store_partition,
+            committed_offset=0,
```

Comment:
Why `committed_offset` is `0` here?
`0` means that there's one processed message